### PR TITLE
use puppetlabs-{hocon, puppet_authorization} for auth.conf

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: spec/classes/puppet_server_puppetserver_spec.rb [hunk 1]
```diff
@@ -502,55 +514,62 @@
...
+
           it {
-            should contain_file('/etc/custom/puppetserver/conf.d/auth.conf').
-              without_content(%r{^(\ *)path: "/puppet/experimental"$})
-          }
-        end
+            should contain_puppet_authorization__rule('puppetlabs experimental').
+              with_ensure('absent').
+              with_path('/etc/custom/puppetserver/conf.d/auth.conf')
+          }        end
```
Comment (ekohl) [inline spec/classes/puppet_server_puppetserver_spec.rb]: Looks like a newline was accidentally removed.

File: manifests/server/puppetserver.pp [hunk 2]
```diff
@@ -251,9 +251,289 @@
...
+    path               => $auth_conf,
+    require            => File[$auth_conf],
+  }
+
+  $auth_conf_setting_ensure = $server_ca ? {
+    true    => present,
+    default => absent,
+  }
+
+  if $server_ca_auth_required == false {
```
Comment (ekohl) [inline manifests/server/puppetserver.pp]: To me it makes more sense to have ```if $server_ca_auth_required {```

File: spec/classes/puppet_server_puppetserver_spec.rb (hunk 1 above)
Comment (mmoll) [reply spec/classes/puppet_server_puppetserver_spec.rb]: fixed

File: manifests/server/puppetserver.pp (hunk 2 above)
Comment (mmoll) [reply manifests/server/puppetserver.pp]: done

**Comment by ekohl** [comment]
I've added the PR to https://etherpad.openstack.org/p/puppet-community-triage and hopefully that will get it attention.

**Comment by mmoll** [comment]
I re-kicked the tests, now that 0.4.0 is out

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**